packages/apps/reputation-oracle/server/src/common/guards/signature.auth.ts

```diff
@@ -52,21 +52,25 @@ export class SignatureAuthGuard implements CanActivate {
       data.chain_id,
       data.escrow_address,
     );
+    if (!escrowData) {
+      throw new HttpException('Escrow not found', HttpStatus.BAD_REQUEST);
+    }
+
     if (
       this.authorizedSignerRoles.includes(AuthSignatureRole.JOB_LAUNCHER) &&
-      escrowData.launcher.length
+      escrowData?.launcher?.length
     ) {
       oracleAdresses.push(escrowData.launcher);
     }
     if (
       this.authorizedSignerRoles.includes(AuthSignatureRole.EXCHANGE_ORACLE) &&
-      escrowData.exchangeOracle?.length
+      escrowData?.exchangeOracle?.length
     ) {
       oracleAdresses.push(escrowData.exchangeOracle);
     }
     if (
       this.authorizedSignerRoles.includes(AuthSignatureRole.RECORDING_ORACLE) &&
-      escrowData.recordingOracle?.length
+      escrowData?.recordingOracle?.length
```

Comment:
nit: make it as in Fortune?
```suggestion
      escrowData.recordingOracle
```